README.md

```diff
@@ -234,100 +240,350 @@ The complete list of transformations supported and their usage in ImageKit can b
 | effectGray | e-grayscale |
 | original | orig |
 
-
-
 ### File Upload
 
 The SDK provides a simple interface using the `.upload()` method to upload files to the ImageKit Media Library. It accepts all the parameters supported by the [ImageKit Upload API](https://docs.imagekit.io/api-reference/upload-file-api/server-side-file-upload).
 
 The `upload()` method requires at least the `file` and the `fileName` parameter to upload a file. You can pass other parameters supported by the ImageKit upload API using the same parameter name as specified in the upload API documentation. For example, to specify tags for a file at the time of upload, use the `tags` parameter as specified in the [documentation here](https://docs.imagekit.io/api-reference/upload-file-api/server-side-file-upload).
 
 Sample usage
+
 ```cs
-ImagekitResponse resp = await imagekit
-    .FileName("my_file_name.jpg")
-    .UploadAsync(<fullPath|url|base_64|binary>);
+ResponseMetaData resp = await imagekit.UploadAsync(<fullPath|url|base_64|binary>,"my_file_name.jpg");
 ```
 
-**Note**: Upload argument can be a local fullPath or URL or byte array (byte[]) or Base64String of a file.
-
+**Note**: Upload argument can be a local fullPath or URL or byte array (byte\[\]) or Base64String of a file.
 
 ### File Management
 
 The SDK provides a simple interface for all the [media APIs mentioned here](https://docs.imagekit.io/api-reference/media-api) to manage your files.
 
-**1. List & Search Files**
+**1\. List & Search Files**
 
 Accepts an object specifying the parameters to be used to list and search files. All parameters specified in the [documentation here](https://docs.imagekit.io/api-reference/media-api/list-and-search-files) can be passed as it is with the correct values to get the results.
 
 ```cs
-List<ListAPIResponse> resp = await imagekit
-    .Skip(0)
-    .Limit(10)
-    .ListFilesAsync();
+GetFileListRequest ob = new GetFileListRequest();
+ob.limit = 10;
+ob.skip = 0;
+ResponseMetaData resp = await imagekit.GetFileListRequestAsync(ob);
 ```
 
-You can also use advanced [searching](https://docs.imagekit.io/api-reference/media-api/list-and-search-files#advanced-search-queries) and sorting using `SearchQuery` and `Sort` function.
+**2\. Get File Details**
+
+Accepts the file ID and fetches the details as per the [API documentation here](https://docs.imagekit.io/api-reference/media-api/get-file-details).
 
 ```cs
-List<ListAPIResponse> resp = await imagekit
-    .Skip(0)
-    .Limit(3)
-    .Sort("DESC_SIZE")
-    .SearchQuery("tags IN [\"tag1\"]")
-    .ListFilesAsync();
+ResponseMetaData resp = await imagekit.GetFileDetailsAsync(fileId);
```

Comment:
`Response` should be a base class, and all methods should return a specific class object which inherits from the `Response` base class. The specific classes should have specific properties.
`Response` class should have `raw` and `statusCode` properties. This should be in readme docs.